### PR TITLE
Player ids are generated from the backend

### PR DESCRIPTION
**Pull request by Eik-S**
- player ids are generated analogue to lobby ids
- frontend does not need to create objects for adding players

### CODE REVIEW

File: game/src/main/java/io/castles/game/GameLobby.java [hunk 1]
```diff
@@ -44,6 +47,26 @@ public void removePlayer(Player player) {
         }
     }
 
+    public void removePlayer(UUID playerId) {
+        Player playerToRemove = getPlayerById(playerId);
+        boolean removed = this.players.remove(playerToRemove);
+        if (!removed) {
+            throw new IllegalArgumentException(String.format("Player %s was not found in the list of players %s", playerToRemove, players));
+        }
```
Comment (soerenreichardt) [inline game/src/main/java/io/castles/game/GameLobby.java:55]:
```suggestion
        removePlayer(playerToRemove)
```
Either this or remove the other `removePlayer` method.

File: game/src/main/java/io/castles/game/GameLobby.java [hunk 2]
```diff
@@ -44,6 +47,26 @@ public void removePlayer(Player player) {
...
+    }
+
+    private Player getPlayerById(UUID playerId) {
+         List<Player> matchingPlayers = this.players.stream()
+             .filter(player -> player.getId().equals(playerId))
```
Comment (soerenreichardt) [inline game/src/main/java/io/castles/game/GameLobby.java:60]: could be `findFirst` instead of filter, but that's nothing important

**Review by soerenreichardt — APPROVED** [review]
👍

File: game/src/main/java/io/castles/game/GameLobby.java [hunk 3]
```diff
@@ -49,21 +46,17 @@ public void removePlayer(Player player) {
...
-             throw new IllegalArgumentException(String.format("Player with if %s was not found in the list of players %s", playerId, players));
-
-         return matchingPlayers.get(0);
+        try {
+            return this.players.stream()
+                    .filter(player -> player.getId().equals(playerId))
+                    .findFirst().get();
+        } catch (NoSuchElementException e) {
+            throw new NoSuchElementException(String.format("Player with if %s was not found in the list of players %s", playerId, players));
+        }
```
Comment (Eik-S) [inline game/src/main/java/io/castles/game/GameLobby.java:59]: Is this usage of `.findFirst()` correct?

Comment (soerenreichardt) [reply game/src/main/java/io/castles/game/GameLobby.java]: Never saw it that way :) usually you work with the optional value by either calling `isPresent()` or chain an `orElseThrow` like I did in the following commit

**Pull request merged**